embeddings/config/flair_config.py

```diff
@@ -0,0 +1,175 @@
+from abc import ABC
+from dataclasses import dataclass, field
+from types import MappingProxyType
+from typing import Any, ClassVar, Dict, Set, Tuple, Union
+
+from embeddings.config.base_config import AdvancedConfig, BasicConfig
+from embeddings.data.io import T_path
+from embeddings.utils.utils import read_yaml
+
+
+@dataclass
+class FlairSequenceLabelingConfigKeys:
+    TASK_MODEL_KEYS: ClassVar[Set[str]] = {
+        "hidden_size",
+        "use_crf",
+        "use_rnn",
+        "rnn_type",
+        "rnn_layers",
+        "dropout",
+        "word_dropout",
+        "locked_dropout",
+        "reproject_embeddings",
+    }
+
+
+@dataclass
+class FlairTextClassificationConfigMapping:
+    TASK_MODEL_KEYS_MAPPING: ClassVar[MappingProxyType[str, Any]] = MappingProxyType(
+        {
+            "FlairDocumentCNNEmbeddings": {
+                "cnn_pool_kernels",
+                "dropout",
+                "word_dropout",
+                "reproject_words",
+            },
+            "FlairDocumentRNNEmbeddings": {
+                "hidden_size",
+                "rnn_type",
+                "rnn_layers",
+                "bidirectional",
+                "dropout",
+                "word_dropout",
+                "reproject_words",
+            },
+            "FlairTransformerDocumentEmbedding": {"dynamic_pooling", "dynamic_fine_tune"},
+            "FlairDocumentPoolEmbedding": {"static_pooling", "static_fine_tune_mode"},
+        }
+    )
+
+
+@dataclass
+class FlairBasicConfig(BasicConfig):
+    TASK_TRAIN_KEYS: ClassVar[Set[str]] = {
+        "learning_rate",
+        "mini_batch_size",
+        "max_epochs",
+    }
+    document_embedding_cls: str = "FlairDocumentPoolEmbedding"
+    learning_rate: float = 1e-3
+    mini_batch_size: int = 32
+    max_epochs: int = 20
+
+    task_model_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+    task_train_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+    load_dataset_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
```

Comment:
Do we need `load_dataset_kwargs` keys here? I think that model/task configs should be separated from the dataset config